src/game/common/sparsematchfinder.h

```diff
@@ -17,18 +17,97 @@
 #include "always.h"
 #include <map>
 
-template<typename K, typename T> class SparseMatchFinder
+template<typename T, typename K> class SparseMatchFinder
 {
     class MapHelper
     {
     public:
         bool operator()(const K &left, const K &right) const { return left == right; }
     };
 
+    static int Count_Condition_Intersection(const K &key1, const K &key2) { return key1.Count_Intersection(key2); }
+    static int Count_Condition_Inverse_Intersection(const K &key1, const K &key2)
+    {
+        return key1.Count_Inverse_Intersection(key2);
+    }
+
 public:
-    // TODO implement more wrappers as they are determined required.
-    T &operator[](const K &key) { return m_bestMatches[key]; }
```

Comment:
Have been unable to find an operator[] for SparseMatchFinder, hence the removal.